### PR TITLE
Starting off the block

### PR DESCRIPTION
**Pull request by mattdodge**

### CODE REVIEW

**Comment by hansmosh** [comment]
@mattdodge I never have access to the repos you create.

**Comment by mattdodge** [comment]
Ugh, so annoying, should be fixed now

File: slack_block.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    ExpressionProperty, ObjectProperty, PropertyHolder
+
+
+class BotInformation(PropertyHolder):
+
+    name = ExpressionProperty(title='Name', default='')
+    icon = ExpressionProperty(
+        title='Icon (URL or emoji)', default=':smiley_cat:')
+
+    def get_bot_details(self, signal):
```
Comment (hansmosh) [inline slack_block.py:15]: This is the first time I've seen a helper method defined in a PropertyHolder like this. Cool.

File: slack_block.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+@Discoverable(DiscoverableType.block)
+class Slack(Block):
+
+    """ Send messages to a slack channel as a bot """
+
+    version = VersionProperty('0.1.0')
+    api_token = StringProperty(
+        title='Slack API Token', default='[[SLACK_API_TOKEN]]')
+    channel = ExpressionProperty(title='Slack Channel', default='{{$channel}}',
```
Comment (hansmosh) [inline slack_block.py]: I have started to prefer to put spaces in my expressions. `{{ $channel }}` 

It's just a personal preference but we should talk about this and be more consistent.

File: slack_block.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        """ Send a message to a Slack channel
+
+        Pass additional kwargs for additional parameters to be sent to the
+        Slack API call. These parameters can be found here:
+        https://api.slack.com/methods/chat.postMessage
+        """
+        self._logger.debug("Sending message {} to channel {}".format(
+            message, to_channel))
+
+        resp = self._slacker.chat.post_message(to_channel, message, **kwargs)
```
Comment (hansmosh) [inline slack_block.py:83]: What about notifying this response as a signal? Or enrich the incoming signal? We can also add this in the next iteration of the block.

File: tests/test_slack_block.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+            'channel': 'This is my channel'
+        })])
+        blk.stop()
+
+        mock.assert_called_once_with(
+            'This is my channel', 'This is my message',
+            icon_url='http://ICONURL', username='Bot Name')
+
+    def test_no_message(self):
+        """ Tests message is not sent if it can't be found on the signal """
```
Comment (hansmosh) [inline tests/test_slack_block.py:67]: Is it overkill to do the same for channel too? 

With expression properties, I like to have tests where the expression fails, to make sure that is handled gracefully. ex. `{{ message }}` without the `$`

File: slack_block.py (hunk 3 above)
Comment (mattdodge) [inline slack_block.py:83]: Yeah, we should enrich it probably, I'll add an issue for this

Comment (mattdodge) [inline slack_block.py:83]: #2

**Pull request merged**